build.rs

```diff
@@ -123,21 +137,23 @@ fn build(sdk_path: Option<&str>, target: &str) {
     // See https://github.com/rust-lang/rust-bindgen/issues/1211
     // Technically according to the llvm mailing list, the argument to clang here should be
     // -arch arm64 but it looks cleaner to just change the target.
-    let target = if target == "aarch64-apple-ios" {
-        "arm64-apple-ios"
-    } else if target == "aarch64-apple-darwin" {
-        "arm64-apple-darwin"
-    } else {
-        target
+    // The full list of clang targtes may be:
+    // https://github.com/llvm/llvm-project/blob/7476c20c481cbccbdb89139fb94620e083015932/llvm/include/llvm/BinaryFormat/MachO.def#L123-L138
+    let clang_target = match target {
+        "aarch64-apple-ios" => "arm64-apple-ios",
+        "aarch64-apple-visionos" => "arm64-apple-xros",
+        "aarch64-apple-visionos-sim" => "aarch64-apple-xros-simulator",
+        "aarch64-apple-darwin"  => "arm64-apple-darwin",
+        target => target,
```

Comment:
I am unsure that this link is fully relevant/correct. Getting the target triple of `aarch64-apple-xros-simulator` took a bit of brute force... ~now that I look at it, I don't know that I tried `arm64-aple-xros-simulator` as a target.~ This passes with both `aarch64-apple-xros-simulator` and `arm64-apple-xros-simulator`. Neither are close to the rust target triples so I have no idea what to use.

Reply:
there was a conversation about this at clang and rust-lang.
Apple originally rolled out `xros` targets in llvm and clang. Later they released `visionOS` branding and that's what Xcode uses, but Xcode translates it to `xros` and sends it to clang. See my #102 notes.